07.ls_object/lib/ls_manager.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require 'date'
+require_relative 'detailed_file_manager'
+require_relative 'file_manager'
+
+class LsManager
+  def initialize(options, path = nil)
+    @path = path
+    @is_dotmatch = options['a']
+    @is_reversed = options['r']
+    @is_detailed = options['l']
+  end
+
+  def output
+    files = get_files_from_path(@path, @is_dotmatch)
+    @is_detailed ? output_detail(files) : output_without_detail(files)
+  end
+
+  private
+
+  def get_files_from_path(path, is_dotmatch)
+    dotmatch_pattern = is_dotmatch ? File::FNM_DOTMATCH : 0
+    files = if path.nil?
+              Dir.glob('*', dotmatch_pattern)
+            elsif FileTest.directory? path
+              Dir.glob(File.join(path, '*'), dotmatch_pattern)
+            elsif FileTest.file? path
+              [path]
+            else
+              raise ArgumentError "ls: #{ARGV[0]}: No such file or directory"
+            end
+    @is_reversed ? files.reverse! : files
+    FileManager.new(files)
+  end
+
+  def output_detail(files)
+    detailed_files = files.get_detailed_files(@path)
+    total_blocks = files.get_total_blocks(detailed_files)
+    puts "total #{total_blocks}"
+    detailed_files.map do |detailed_file|
+      types, permission, nlink, user, group, size, mtime, file_name = detailed_file[1..]
+      puts [types, permission, nlink, user, group, size, mtime, file_name].join
```

Comment:
特に中間処理をしないなら

```suggestion
      puts detailed_file[1..].join
```

で十分そうですかね

Reply:
おっしゃる通りですね🙇‍♂️
修正しました🙇‍♂️